### PR TITLE
changing callback_path to callback_url to account for relative root url

### PR DESCRIPTION
**Pull request by angelkbrown**
When an app is deployed to a subdirectory of the webroot, so that the root url is like http://host/subdirectory, the authentication callback fails, since callback_path is passed as the action to the form, which causes the user to be directed to http://host/auth/ldap/callback. 

This fix passes callback_url as the action to the form so that the form action is the full url: http://host/subdirectory/auth/ldap/callback for applications deployed to a subdirectory, or http://host/auth/ldap/callback for applications deployed in the host's webroot.

### CODE REVIEW

**Comment by aldanor** [comment]
Has anyone figured how to resolve this?

Including the sub URI in the `:callback_path` seems wrong, because then `on_callback_path?` would fail (as it matches against it). So the correct solution would indeed be to use `callback_url`.

**Comment by aldanor** [comment]
TL;DR: this PR does look like a correct solution, are there are any reasons not to merge it in?

**Comment by jmccann** [comment]
I just ran into this.  Tried to work around it by forcing the callback_path but `env['omniauth.auth']` is nil.  Seems like this is a known issue when you set `:callback_path` as is suggested as the workaround for this issue.

https://github.com/intridea/omniauth/issues/767

**Comment by aldanor** [comment]
@jmccann 

I was able to make it work by doing something like this:

```ruby
module OmniAuthLDAPExt
    def request_phase
        @callback_path = nil  # latest version caches callback path
        path = options[:callback_path]
        options[:callback_path] = callback_url
        form = super
        options[:callback_path] = path
        form
    end
end

module OmniAuth
    module Strategies
        class LDAP
            prepend OmniAuthLDAPExt
        end
    end
end
```

**Comment by jmccann** [comment]
@aldanor Thanks!  I'm testing monkeypatching in the fix right now doing the following as part of my setup:

```ruby
module OmniAuth
  module Strategies
    class LDAP
      def request_phase
        OmniAuth::LDAP::Adaptor.validate @options
        f = OmniAuth::Form.new(:title => (options[:title] || "LDAP Authentication"), :url => callback_url)
        f.text_field 'Login', 'username'
        f.password_field 'Password', 'password'
        f.button "Sign In"
        f.to_response
      end
    end
  end
end
```

**Comment by aldanor** [comment]
@jmccann it's pretty much the same, I just wanted to avoid copying/pasting all the form code, hence using `prepend` / `super`.

**Comment by opensorceror** [comment]
This issue still persists...looks like this PR is not going to get merged anytime soon?

**Comment by opensorceror** [comment]
I was able to solve it using the following Nginx rewrite rule as suggested by @lud0vicb [here](https://github.com/jonmbake/discourse-ldap-auth/issues/8#issue-160633314):
```
rewrite ^/auth/ldap/(.*)$ /forum/auth/ldap/$1 last
```
